### DISCUSSION TITLE
how to create a moving and flashing star

### DISCUSSION THREAD

**Discussion by ethanruiyu** [pmndrs/react-spring#1715] [Support] [answered]
Hi, everyone.
  I just started using react spring，I'm not familiar with its usage.
  I have a question. I don't know what to do

  demo: https://codesandbox.io/s/hardcore-rosalind-0tjvf?file=/src/App.js

  when not moving, it flashing, but when moving, flashing(opacity) not work

**Comment by joshuaellis** [comment, marked as answer]
You were updating the state to change the values of your spring, which is okay if its one value or another, but I wouldn't advising doing it like you were doing it – updating in an interval.

You might find it easier to use the [imperative API](https://react-spring.io/common/imperatives-and-refs#api-methods) we provide as can be seen in this code sandbox – https://codesandbox.io/s/hopeful-pike-kvsju?file=/src/App.js
